aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1411,6 +1411,15 @@ Tensor& mean_out(const Tensor& self, DimnameList dim,
   return at::mean_out(result, self, dimnames_to_positions(self, dim), keepdim, opt_dtype);
 }
 
+Tensor& mean_dtype_out(const Tensor &self, std::optional<ScalarType> dtype, Tensor& result) {
+  TORCH_CHECK(
+    canCast(self.scalar_type(), result.scalar_type()),
```

Comment:
oh, the test is failing for you because:

(1) `torch.mean(..., out=)` can desugar to 1 of 2 overloads (the old one, and the new one you added)

(2) the OpInfo tests use `torch.mean`, so they run on **both** out= overloads

(3) you've added the error for this new overload, but the old mean doesn't have the error

Reply:
The error seems good to have. It would also be nice to add the error to the other overload (5 lines above this one), so we can avoid skipping the test. Although there is a tiny BC risk (we would probably just want to label the module BC breaking, so we can mention it in the release notes).

I think given the tiny BC risk, I'm ok with either option:

(1) add the error to both overloads and fix the test

(2) keep the error just in the new op, but leave the test xskipp'd.

Reply:
ok had to go with (2) since (1) doesn't fix the issue. I'm having a hard time to tell if the test is actually hitting the kernels. I added the error message in the skip line.